.github/workflows/ci.yml

```diff
@@ -35,19 +35,19 @@ jobs:
     runs-on: ubuntu-20.04
 
     steps:
-      - uses: actions/checkout@v2
+      - uses: actions/checkout@v4
 
       - name: Tune Runner VM
         uses: ./.github/actions/tune-runner-vm
 
-      - name: Set up JDK 11
-        uses: actions/setup-java@v2
+      - name: Set up JDK 17
+        uses: actions/setup-java@v4
         with:
           distribution: 'temurin'
-          java-version: 11
+          java-version: 17
 
       - name: Setup Gradle
-        uses: gradle/gradle-build-action@v2
+        uses: gradle/actions/setup-gradle@v3
 
       - name: Check with Gradle
```

Comment:
It would be great to add some runtime verification w/ Java 8 (not compile, just runtime) to ensure this is still good w/ JDK8. We do not currently have samples to run though so not sure what the best approach here would be.

Reply:
Created https://github.com/apache/pulsar-client-reactive/issues/182 to follow up